mesh_generation/generate_mesh.py

```diff
@@ -27,79 +27,19 @@
 # =========================================================================================
 
 import os
-import io
-import hashlib
-import warnings
-import subprocess
 from datetime import datetime
 
 import numpy as np
 import xarray as xr
 import pandas as pd
 
+from pathlib import Path
+import sys
+path_root = Path(__file__).parents[1]
+sys.path.append(str(path_root))
 
-def get_git_url(file):
-    """
-    If the provided file is in a git repo, return the url to its most recent commit remote.origin.
-    """
-    dirname = os.path.dirname(file)
-
-    try:
-        url = subprocess.check_output(
-            ["git", "-C", dirname, "config", "--get", "remote.origin.url"]
-        ).decode("ascii").strip()
-        url = url.removesuffix('.git')
-    except subprocess.CalledProcessError:
-        return None
-
-    if url.startswith("git@github.com:"):
-        url = f"https://github.com/{url.removeprefix('git@github.com:')}"
-
-    top_level_dir = subprocess.check_output(
-        ["git", "-C", dirname, "rev-parse", "--show-toplevel"]
-    ).decode("ascii").strip()
-    rel_path = file.removeprefix(top_level_dir)
-
-    hash = subprocess.check_output(
-        ["git", "-C", dirname, "rev-parse", "HEAD"]
-    ).decode("ascii").strip()
-
-    return f"{url}/blob/{hash}{rel_path}"
-
-def git_status(file):
-    """
-    Return the git status of the file. Returns:
-    - "unstaged" if the file has unstaged changes
-    - "uncommitted" if the file has uncommited changes,
-    - "unpushed" if the repo has unpushed commits
-    - None otherwise
-    """
-    dirname = os.path.dirname(file)
-    status = subprocess.check_output(
-        ["git", "-C", dirname, "status", file]
-    ).decode("ascii").strip()
-
-    if "Changes not staged for commit" in status:
-        return "unstaged"
-    elif "Changes to be committed" in status:
-        return "uncommitted"
-    elif "Your branch is ahead" in status:
-        return "unpushed"
-    else:
-        return None
-
-def md5sum(path):
-    """
-    Return the md5 hash of a provided file, reading in chunks to reduce memory usage for
-    large files.
-    From https://stackoverflow.com/a/40961519
-    """
-    length = io.DEFAULT_BUFFER_SIZE
-    md5 = hashlib.md5()
-    with io.open(path, mode="rb") as fd:
-        for chunk in iter(lambda: fd.read(length), b''):
-            md5.update(chunk)
-    return md5.hexdigest()
+from tools.git import get_created_str
+from tools.md5sum import *
```

Comment:
Could we do this with a relative import instead to avoid having to add to `PYTHONPATH`?

Reply:
Yeah, I think we'd need to add some `__init__.py`s around the place and that probably won't help with clarity... Let's stick with adding to `PYTHONPATH` but name the new module something we can be pretty sure is unique